.github/workflows/glog.yml

```diff
@@ -0,0 +1,102 @@
+name: glog
+
+on:
+  push:
+    branches:
+      - '*'
+    path:
+      - 'instrumentation/glog/**'
+      - '.github/workflows/glog.yml'
+  pull_request:
+    branches: [main]
+    paths:
+      - 'instrumentation/glog/**'
+      - '.github/workflows/glog.yml'
+
+jobs:
+  cmake_linux:
+    name: CMake Linux
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout googletest
+        uses: actions/checkout@v3
+        with:
+          repository: "google/googletest"
+          ref: "release-1.12.1"
+          path: "googletest"
+          submodules: "recursive"
+      - name: checkout glog
+        uses: actions/checkout@v3
+        with:
+          repository: "google/glog"
+          # TODO: upgrade to v0.7.0 and fix deprecation warnings
+          ref: "v0.6.0"
+          path: "glog"
+          submodules: "recursive"
+      - name: checkout opentelemetry-cpp-contrib
+        uses: actions/checkout@v3
+        with:
+          path: opentelemetry-cpp-contrib
+          # submodules: "recursive"
+      - name: checkout opentelemetry-cpp
+        uses: actions/checkout@v3
+        with:
+          repository: "open-telemetry/opentelemetry-cpp"
+          ref: "v1.14.2"
+          path: "opentelemetry-cpp"
+          submodules: "recursive"
+      - name: setup dependencies
+        run: |
+          sudo apt update -y
+          sudo apt install -y --no-install-recommends --no-install-suggests \
+            build-essential \
+            cmake \
+            ninja-build \
+            libssl-dev \
+            libcurl4-openssl-dev \
+            libprotobuf-dev \
+            protobuf-compiler \
+            libgmock-dev \
+            libgtest-dev \
```

Comment:
Is the source check our needed as `libgtest-dev` is installed?

Reply:
Yes, a source checkout for googletest 1.12 is required, because libgtest-dev only installs 1.11.

Installing both 1.11 and 1.12 also tests this line:

```
find_package(GTest 1.12 REQUIRED)
```

to make sure asking for 1.12 discards the default libgtest-dev.